react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ''
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit(event) {
+    console.log('state:', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        Email<input onChange={this.handleChange} type="text"></input><button>Sign Up</button>
```

Comment:
The `input` should have a `value` attribute set to the value from state that is controlling the input, in this case `this.state.email`